ci/unit/test_main_node.sh

```diff
@@ -22,9 +22,29 @@ sudo ln -sf "$SL_PREFIX/share/spdk/scripts/setup.sh" /usr/share/spdk/scripts
 sudo ln -sf "$SL_PREFIX/share/spdk/scripts/common.sh" /usr/share/spdk/scripts
 sudo ln -s "$SL_PREFIX/include"  /usr/share/spdk/include
 
-# set CMOCKA envs here
-export CMOCKA_MESSAGE_OUTPUT=xml
-export CMOCKA_XML_FILE="$DAOS_BASE"/test_results/%g.xml
 cd "$DAOS_BASE"
-IS_CI=true OLD_CI=false utils/run_test.sh
-./utils/node_local_test.py all
+if [ "$WITH_VALGRIND" = "memcheck" ]; then
+    ls results-*-memcheck.xml
+    ls test_results/results-*-memcheck.xml
+    ls results-*-memcheck.xml | wc
+    ls test_results/results-*-memcheck.xml | wc
+    rm results-*-memcheck.xml
+    rm test_results/results-*-memcheck.xml
+
+    # run_test.sh with valgrind memcheck
+    IS_CI=true OLD_CI=false RUN_TEST_VALGRIND=memcheck utils/run_test.sh
+
+    ls results-*-memcheck.xml
+    ls test_results/results-*-memcheck.xml
+    ls results-*-memcheck.xml | wc
+    ls test_results/results-*-memcheck.xml | wc
+    # Remove DAOS_BASE from memcheck xml results
+    find test_results -maxdepth 1 \
```

Comment:
(lint) Use -print0/-0 or -exec + to allow for non-alphanumeric filenames. [SC2038]